### PR TITLE
Created database setup

### PR DESCRIPTION
**Pull request by mmubarak0**
# Refactored database setup
This is rework of the latest database fixed some bugs and added missing tables.

# Bugs Fixed
- [x] Reworked the database structure
- [x] Ensured all models inherit properly from the BaseModel
- [x] Corrected the Base import from the database, not the base model
- [x] Added all newly implemented features
- [x] no breaking changes has been introduced.
- [x] all previous and new models are tested.
- [x] The readme file has been re written completely

### CODE REVIEW

**Comment by mmubarak0** [comment]
@Dev-wonderful Please can you review this ?

File: api/db/database.py [hunk 1]
```diff
@@ -28,8 +29,11 @@ def get_db_engine(test_mode: bool = False):
             return create_engine(
                 DATABASE_URL, connect_args={"check_same_thread": False}
             )
-    elif DB_TYPE == "postgresql":
-        DATABASE_URL = f"postgresql://{DB_USER}:{DB_PASSWORD}@{DB_HOST}:{DB_PORT}/{DB_NAME}"
+    else:
+        if DB_DRIVER:
+            DATABASE_URL = f"{DB_TYPE}+{DB_DRIVER}://{DB_USER}:{DB_PASSWORD}@{DB_HOST}:{DB_PORT}/{DB_NAME}"
+        else:
+            DATABASE_URL = f"{DB_TYPE}://{DB_USER}:{DB_PASSWORD}@{DB_HOST}:{DB_PORT}/{DB_NAME}"
```
Comment (bolexs) [inline api/db/database.py:36]: You are doing the same thing

**Review by bolexs — COMMENTED** [review]
I believe you should discuss this changes with your team lead and database team. As to why this changes are needed or necessary

File: api/db/database.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env python3
 """ The database module
 """
-from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import declarative_base
```
[the remark this answers is not in the document]
Comment (mmubarak0) [reply api/db/database.py:4]: There are no issues, but the warning appears every time I run PyTest. Below is the warning message that is reproduced.
image

File: api/db/database.py (hunk 1 above)
Comment (mmubarak0) [reply api/db/database.py:36]: I thought It might be more portable in this manner, but perhaps it's best not to alter that section.

Comment (mmubarak0) [reply api/db/database.py:36]: Should I revert this part ?

**Comment by Dev-wonderful** [comment]
There are a lot of changes that could break everyone's setup at this moment

**Pull request closed without merging**